mongodbatlas/resource_mongodbatlas_private_endpoint_regional_mode.go

```diff
@@ -0,0 +1,183 @@
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	errorPrivateEndpointRegionalModeRead    = "error reading MongoDB Group `%s Private Endpoints Regional Mode: %s"
+	errorPrivateEndpointRegionalModeSetting = "error setting `%s` on MongoDB Group `%s` Private Endpoints Regional Mode: %s"
+)
+
+func resourceMongoDBAtlasPrivateEndpointRegionalMode() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMongoDBAtlasPrivateEndpointRegionalModeCreate,
+		ReadContext:   resourceMongoDBAtlasPrivateEndpointRegionalModeRead,
+		UpdateContext: resourceMongoDBAtlasPrivateEndpointRegionalModeUpdate,
+		DeleteContext: resourceMongoDBAtlasPrivateEndpointRegionalModeDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMongoDBAtlasPrivateEndpointRegionalModeImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceMongoDBAtlasPrivateEndpointRegionalModeCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	d.SetId(d.Get("project_id").(string))
+	err := resourceMongoDBAtlasPrivateEndpointRegionalModeUpdate(ctx, d, meta)
+
+	if err != nil {
+		return err
+	}
+
+	return resourceMongoDBAtlasPrivateEndpointRegionalModeRead(ctx, d, meta)
+}
+
+func resourceMongoDBAtlasPrivateEndpointRegionalModeRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+
+	projectID := d.Id()
+
+	setting, resp, err := conn.PrivateEndpoints.GetRegionalizedPrivateEndpointSetting(context.Background(), projectID)
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(fmt.Errorf(errorPrivateEndpointRegionalModeRead, projectID, err))
+	}
+
+	if err := d.Set("enabled", setting.Enabled); err != nil {
+		return diag.FromErr(fmt.Errorf(errorPrivateEndpointRegionalModeSetting, "enabled", projectID, err))
+	}
+
+	return nil
+}
+
+func resourceMongoDBAtlasPrivateEndpointRegionalModeUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+
+	projectID := d.Id()
+	enabled := d.Get("enabled").(bool)
+
+	_, resp, err := conn.PrivateEndpoints.UpdateRegionalizedPrivateEndpointSetting(ctx, projectID, enabled)
+	if err != nil {
+		if resp != nil && resp.Response.StatusCode == 404 {
+			return nil
+		}
+
+		return diag.FromErr(fmt.Errorf(errorPrivateLinkEndpointsDelete, projectID, err))
+	}
+
+	log.Println("[INFO] Waiting for MongoDB Private Endpoints Connection to be destroyed")
```

Comment:
```suggestion
	log.Println("[INFO] Waiting for MongoDB Private Endpoints Connection to be updated")
```

Reply:
I ended up changing the wording here a bit, but yes "deleted" was obviously incorrect. Thank you for the call out!